lib/load-changelog-config.js

```diff
@@ -25,19 +25,13 @@ export default async ({ preset, config, parserOpts, writerOpts, presetConfig },
 
   if (preset) {
     const presetPackage = `conventional-changelog-${preset.toLowerCase()}`;
-    loadedConfig = importFrom.silent(__dirname, presetPackage) || importFrom(cwd, presetPackage);
+    loadedConfig = await (importFrom.silent(__dirname, presetPackage) || importFrom(cwd, presetPackage))(presetConfig);
   } else if (config) {
-    loadedConfig = importFrom.silent(__dirname, config) || importFrom(cwd, config);
+    loadedConfig = await (importFrom.silent(__dirname, config) || importFrom(cwd, config))();
   } else {
-    loadedConfig = conventionalChangelogAngular;
+    loadedConfig = await conventionalChangelogAngular();
   }
 
-  loadedConfig = await (typeof loadedConfig === "function"
-    ? isPlainObject(presetConfig)
-      ? loadedConfig(presetConfig)
-      : promisify(loadedConfig)()
-    : loadedConfig);
-
```

Comment:
since this took me a bit to understand, capturing the reason this is being removed. with the previous preset format, most presets returned an object, but the conventionalcommits preset returned a function that accepted config.

the change that resulted in the breaking changes across the conventional-changelog presets was an effort to [unify the interface across those presets](https://github.com/conventional-changelog/conventional-changelog/pull/1045). therefore, we can instead pass the config above and remove the handling of the different potential return shapes. nice work @dangreen!

Reply:
i figured this out after cutting the first beta of the commit-analyzer plugin, so will make this change there too soon